### ISSUE TITLE
hang while connecting to custom server via hwplay scheme if you have empty nickname

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmith/hedgewars#588]
```
What steps will reproduce the problem?
1. Associate Hedgewars file extension on Linux
2. Click on (eg) hwplay://m8y.org
3. Wait indefinitely

What is the expected output? What do you see instead?
Instead of hanging in the 'connecting' page it should connect to server! Same 
code, only happens on Linux (why?).

It's also possible to cause this by calling $(hedgewars hwplay://m8y.org) from 
the command line
```

Original issue reported on code.google.com by `vittorio...@gmail.com` on 25 Mar 2013 at 11:21
* Blocking: #580

**Comment by GoogleCodeExporter** [comment]
Original comment by `vittorio...@gmail.com` on 27 Mar 2013 at 10:43
* Now blocking: #580

**Comment by GoogleCodeExporter** [comment]
```
Hm... works for me. FreeBSD, r9477443dd583.
```

Original comment by `unC0Rr` on 21 May 2013 at 7:41

**Comment by GoogleCodeExporter** [comment]
```
Ok, the problem was that I had the nick field empty.

Server:  ("CONNECTED", "Hedgewars server http://www.hedgewars.org/", "1") 
Client:  ("NICK", "") 
Client:  ("PROTO", "44") 
Server:  ("ERROR", "Incorrect command (state: not entered)") 
Server:  ("ERROR", "Incorrect command (state: not entered)") 

Will look at it, but shouldn't block release schedule.
```

Original comment by `vittorio...@gmail.com` on 22 May 2013 at 9:40
* Changed title: **hang while connecting to custom server via hwplay scheme if you have empty nickname**
* Changed state: **Accepted**
* Added labels: **Priority-Low**
* Removed labels: **Priority-Medium**, **Milestone-NextRelease**

**Comment by GoogleCodeExporter** [comment]
```
r45ebf126a5c2 might be onto something for fixing this but we should try to have 
a unified solution for avoiding blank nicknames.
```

Original comment by `vittorio...@gmail.com` on 27 May 2013 at 7:44

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 903f50e85144.
```

Original comment by `kyberneticist@gmail.com` on 27 May 2013 at 2:48
* Changed state: **Fixed**
